tests/posix/fs/src/test_fs_dir.c

```diff
@@ -27,7 +27,8 @@ static int test_mkdir(void)
 		return res;
 	}
 
-	res = open(TEST_DIR_FILE, O_RDWR);
+	res = open(TEST_DIR_FILE, O_CREAT | O_RDWR);
```

Comment:
Please extend the commit message of this commit with something like "A few test functions which assumed O_CREAT behavior are updated to specify it explicitly."

Reply:
Done.